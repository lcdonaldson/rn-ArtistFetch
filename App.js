
import React, {Component} from 'react';
import axios from 'axios';
import {
  SafeAreaView,
  StyleSheet,
  View,
  Text,
  StatusBar,
  Pressable,
  FlatList
} from 'react-native';

import {
  Header,
  LearnMoreLinks,
  Colors,
  DebugInstructions,
  ReloadInstructions,
} from 'react-native/Libraries/NewAppScreen';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      itemList: []
    };
  }
  onPressFunction = () => {
    let artist = 'Elvis'
    axios.get(`https://itunes.apple.com/search?term="${artist}"`)
    .then(response => {
        this.setState({
          itemList: response.data.results
        });
        // console.log("Data", this.state.itemList);
      })
      .catch(error => console.log(error));
  }
  render() {
    {console.log('MYDATA', this.state.itemList)}
    return (
      <>
        <StatusBar barStyle="dark-content" />
        <SafeAreaView>
          <View>
            <View>
              <Pressable 
              style={styles.btnLayout} 
              onPress={this.onPressFunction}
              >
                <Text style={styles.btnText}>Show Records</Text>
              </Pressable> 
            </View>
            <View style={styles.listWrapper}>
              <Text>SANITY TEXT 1</Text>
              <FlatList
                data={this.state.itemList}
                
                renderItem={({item}) => {
                  <Text style={styles.listText}>{item}</Text>;
                }}
                keyExtractor={item => item.data.results}
              />
              <Text>SANITY TEXT 2</Text>
            </View>
          </View>
        </SafeAreaView>
      </>
    );
  }
};

const styles = StyleSheet.create({
  btnLayout: {
    backgroundColor: 'blue', 
    minWidth: 320, 
    padding: 5, 
    alignSelf: 'center', 
    marginTop: 25
  },
  listWrapper: {
    padding: 12,
    marginTop: 25,
    backgroundColor: '#ccc'
  },
  listText: {
    fontSize: 20, 
    color: '#000', 
    width: 200, 
    height: 50
  },
  btnText: {
    fontSize: 20, 
    color: '#fff', 
    textAlign: 'center'
  }
});

