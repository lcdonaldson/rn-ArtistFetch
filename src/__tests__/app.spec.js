import React from 'react';
import App from '../../App';
import renderer from 'react-test-renderer';

describe('Snapshot tests', () => {
    // SNAPSHOT
    it('renders correctly when there are no items', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("handleClick function", () => {
    // handleClick HAPPY PATH
    it('fetches data successfully when clicked', () => {
        //Arrange
        let app = new App();
        const data = {
            item: {
                artistName: 'Elvis',
                trackName: 'Suspicious Minds',
                trackPrice: 1.29,
                primaryGenreName: 'Rock',
                releaseDate: 1611862879397
            },
        };

        //Act
        let spy = jest.spyOn(app, 'handleClick').mockImplementation(() => data);

        //Assert
        expect(app.handleClick()).toBe(data);
        expect(spy).toHaveBeenCalled()
    });
    
    // handleClick ERROR PATH
    it('fetches data erroneously when clicked', () => {
        //Arrange
        let app = new App();
        const data = {
            item: {
                artistName: 'Elvis',
                trackName: 'Suspicious Minds',
                trackPrice: 1.29,
                primaryGenreName: 'Rock',
                releaseDate: 1611862879397
            },
        };

        //Act
        let spy = jest.spyOn(app, 'handleClick').mockImplementation(() => null);

        //Assert
        expect(app.handleClick()).not.toBe(data);
        expect(spy).toHaveBeenCalled()
    });
});

describe('dateFormat function', () => {
    // dateFormat HAPPY PATH
    it('returns the correctly formatted date', () => {
        //Arrange
        let app = new App();
        const date = 'Tue, Sep 24, 2002'
    
        //Act
        let spy = jest.spyOn(app, 'dateFormat').mockImplementation(() => date);
    
        //Assert
        expect(app.dateFormat()).toBe('Tue, Sep 24, 2002');
        expect(spy).toHaveBeenCalled()
    });
    // dateFormat ERROR PATH
    it('does not return the correct format', () => {
        //Arrange
        let app = new App();
        const date = 'Tue, Sep 24, 2002'
    
        //Act
        let spy = jest.spyOn(app, 'dateFormat').mockImplementation(() => 1611862879397);
    
        //Assert
        expect(app.dateFormat()).not.toBe(date);
        expect(spy).toHaveBeenCalled()
    });
})


