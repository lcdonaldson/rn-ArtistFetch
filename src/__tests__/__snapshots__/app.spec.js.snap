// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests renders correctly when there are no items 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View>
    <View>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Artist name"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderBottomColor": "#000000",
            "borderBottomWidth": 1,
            "minHeight": 20,
            "width": 200,
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "green",
            "marginTop": 25,
            "minWidth": 320,
            "padding": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          Show Records
        </Text>
      </View>
    </View>
    <ActivityIndicator
      animating={false}
      color="red"
      hidesWhenStopped={true}
      size="large"
    />
    <View
      style={
        Object {
          "marginTop": 25,
          "padding": 12,
        }
      }
    >
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;
